File: Chapter1/exercises_2/ex6/ex6.cpp
// How to compile: g++ -std=c++11 -o ex6 ex6.cpp

#include <iostream>

using namespace std;

int fibonacci(int n);

int main() {
	int n = 3;
	cout << n << " " << fibonacci(n) << endl;
	n = 7;
	cout << n << " " << fibonacci(n) << endl;
	return 0;
}

int fibonacci(int n) {
	if (n == 1)
		return 1;
	
	if (n == 2)
		return 1;

	return fibonacci(n - 1) + fibonacci(n - 2);
}

File: Chapter1/exercises_2/ex6/ex6.py
def fibonacci(n: int) -> int:
	if n == 1:
		return 1
	elif n == 2:
		return 1
	else:
		return fibonacci(n - 1) + fibonacci(n - 2)

def main():
	print(3, fibonacci(3))
	print(7, fibonacci(7))


# 1 1 2 3 5 8 13
if __name__ == '__main__':
	main()